pkg/trie/inmemory/iterator.go

```diff
@@ -74,9 +74,18 @@ func (t *InMemoryTrieIterator) NextKeyFunc(predicate func(nextKey []byte) bool)
 }
 
 func (t *InMemoryTrieIterator) Seek(targetKey []byte) {
-	t.NextKeyFunc(func(nextKey []byte) bool {
-		return bytes.Compare(nextKey, targetKey) >= 0
-	})
+	var prevEntry *trie.Entry
+	for entry := t.NextEntry(); entry != nil; entry = t.NextEntry() {
+		if bytes.Compare(entry.Key, targetKey) >= 0 {
+			break
+		}
+		prevEntry = entry
+	}
+	if prevEntry != nil {
+		t.cursorAtKey = prevEntry.Key
+		return
+	}
+	t.cursorAtKey = nil
```

Comment:
we can store the key itself, then we don't need to check if it is nil, we just assign to `t.cursorAtKey`
```suggestion
	var prevKey []byte = nil
	for entry := t.NextEntry(); entry != nil; entry = t.NextEntry() {
		if bytes.Compare(entry.Key, targetKey) >= 0 {
			break
		}
		prevEntry = entry.Key
	}
	
	t.cursorAtKey = prevKey
```